stratum/hal/bin/barefoot/docker/start-stratum-container.sh

```diff
@@ -11,6 +11,11 @@ DOCKER_IMAGE_TAG=${DOCKER_IMAGE_TAG:-$SDE_VERSION}
 # Try to load the platform string if not already set.
 if [[ -z "$PLATFORM" ]] && [[ -f "/etc/onl/platform" ]]; then
     PLATFORM=$(cat /etc/onl/platform)
+elif [[ -z "$PLATFORM" ]] && [[ -f "/etc/sonic/sonic-environment" ]]; then
+    source /etc/sonic/sonic-environment
+    PLATFORM=$(echo "$PLATFORM" | sed 's/_/-/g')
```

Comment:
Do we care about "polluting" this script with other variables from `sonic-environment`?

Note: The variables are not exported to subprocesses unless you call `set -o allexport` before `source`. So this really only affects the code in this file.

If we do care, here's one solution:
```suggestion
    PLATFORM=$(source /etc/sonic/sonic-environment; echo $PLATFORM | sed 's/_/-/g')
```

Test:
```bash
$ echo $(source /etc/sonic/sonic-environment; echo $PLATFORM | sed 's/_/-/g')
x86-64-accton-wedge100bf-32x-r0
```